craft_parts/plugins/base.py

```diff
@@ -81,6 +82,20 @@ def set_action_properties(self, action_properties: ActionProperties) -> None:
         self._action_properties = deepcopy(action_properties)
 
 
+class StrictPlugin(Plugin):
+    """Base class for strict plugins.
+
+    Strict plugins will retrieve all dependencies from remote locations during
+    the pull step, allowing the build step to run offline.
+    """
+
+    offline = True
```

Comment:
I think something like `allow_offline_build` would be more descriptive here since being offline depends on changes in the environment (i.e., we're not intrinsically offline, but we will still work if the build step happens on a host that's offline).

Reply:
Check the type instead of using an attrib